### PR TITLE
Expose copy_options

### PR DESCRIPTION
**Pull request by hkdahal**
Currently S3-compatible Cloudflare R2 does not support tagging_directive (https://developers.cloudflare.com/r2/api/s3/api/). This means, we cannot use shrine with tagging_directive. Thus, this Pull Request aims to expose copy_options.

By default, copy_options will have existing value: `{ tagging_directive: "REPLACE" }`

For users like me, we can instantiate S3 store like below until Cloudflare is fully compatible (or just pass additional copy options):

`Shrine::Storage::S3.new(bucket: BUCKET, copy_options: {}, **s3_options),`

This will also solve: https://github.com/shrinerb/shrine/issues/662

### CODE REVIEW

**Comment by dylandamsma** [comment]
Can we get this merged into main?

**Comment by hkdahal** [comment]
Awaiting approval from @janko. Can you please review/approve accordingly for this change? thanks

File: .gitignore [hunk 1]
```diff
@@ -5,3 +5,4 @@ website/build
 coverage/
 node_modules
 .docusaurus
+.idea
```
Comment (pokonski) [inline .gitignore:8]: https://gist.github.com/subfuzion/db7f57fff2fb6998a16c

Comment (hkdahal) [reply .gitignore]: thanks for the feedback @pokonski , reverted the changes in that file.

**Comment by hkdahal** [comment]
@pokonski  - can I get your feedback/approval on this Pull Request please? thanks

**Comment by pokonski** [comment]
@hkdahal sorry, I'm not a maintainer of Shrine, just commented a potential improvement :) I too am waiting for this fix 🤞

**Comment by hkdahal** [comment]
pinging @janko  for approval on this PR

**Comment by thatdom** [comment]
+1

**Comment by janko** [comment]
Thanks for the PR, looks good to me 👍🏻

**Pull request merged**